app/models/form526_submission.rb

```diff
@@ -6,10 +6,82 @@
 
 class Form526Submission < ApplicationRecord
   extend Logging::ThirdPartyTransaction::MethodWrapper
-
+  include AASM
   include SentryLogging
   include Form526ClaimFastTrackingConcern
 
+  # Documentation:
+  # https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/disability/526ez/implementation/form_526_state_machine.md
+  aasm do
+    after_all_transitions :log_status_change
+
+    state :unprocessed, initial: true
+    state :delivered_to_primary, :failed_primary_delivery, :rejected_by_primary,
+          :delivered_to_backup, :failed_backup_delivery, :rejected_by_backup,
+          :in_remediation, :finalized_as_successful, :unprocessable
+
+    # - a submission has been delivered to our happy path
+    # - requires polling to finalize
+    event :deliver_to_primary do
+      transitions to: :delivered_to_primary
+    end
+
+    # - submission failed delivery to primary path for any reason
+    # - requires backup submission or remediation
+    event :fail_primary_delivery do
+      transitions to: :failed_primary_delivery
+    end
+
+    # - a successfully delivered submission has failed 3rd party validations on primary path
+    # - requires backup submission or remediation
+    event :reject_from_primary do
+      transitions to: :rejected_by_primary
+    end
+
+    # - a submission has been delivered to our backup path
+    # - accepts any non-success type state
+    # - MUST accept all the same incoming statuses as fail_backup_delivery
+    # - requires polling to finalize
+    event :deliver_to_backup do
+      transitions to: :delivered_to_backup
+    end
+
+    # - a submission has failed to be delivered to our backup path
+    # - accepts any non-success type state
+    # - MUST accept all the same incoming statuses as fail_backup_delivery
+    # - requires remediation
+    event :fail_backup_delivery do
+      transitions to: :failed_backup_delivery
+    end
+
+    # - a successfully delivered submission has failed 3rd party validations on backup path
+    # - requires remediation
+    event :reject_from_backup do
+      transitions to: :failed_backup_delivery
+    end
+
+    # - Submission has entered a manual remediation flow, e.g. failsafe, paper submission
+    # - requires confirmation of success, e.g. polling or manual confirmation via audit
+    event :begin_remediation do
+      transitions to: :in_remediation
+    end
+
+    # - The only state that means we no longer own completion of this submission
+    # - There is nothing more to do.  E.G.
+    #   - VBMS has accepted and returned the applicable status to us via LH
```

Comment:
Suggest writing out Lighthouse here, to avoid any future confusion regarding meaning.